test/unit/main_test.ts

```diff
@@ -64,13 +75,30 @@ describe('Recognizes invalid TypeScript inputs', () => {
 });
 
 describe('Visitor pattern', () => {
-  it('shows correctly appends "$mangled" to an identifier of a property access expression', () => {
-    expectTranslate('Math.random();').to.equal('Math.random$mangled();');
-    expectTranslate('Class Foo { a: string; constructor() {} b() { this.a = "hello"; } }')
-        .to.equal('Class Foo { a: string; constructor() {} b() { this.a$mangled = "hello"; } }');
-    expectTranslate('for (x in foo.bar) { var y = foo.bar.baz; }')
-        .to.equal('for (x in foo.bar$mangled) { var y = foo.bar$mangled.baz$mangled; }');
-    expectTranslate('var x = foo.baz();').to.equal('var x = foo.baz$mangled();');
+  it('renames identifiers of property declarations and property access expressions', () => {
+    expectTranslate('class Foo { bar: string; constructor() {} baz() { this.bar = "hello"; } }')
+        .to.equal('class Foo { $: string; constructor() {} _() { this.$ = "hello"; } }');
+    expectTranslate('for (var x in foo.bar) { var y = foo.bar.baz; }')
+        .to.equal('for (var x in $._) { var y = $._.a; }');
+    expectTranslate('class Foo {bar: string;} class Baz {bar: string;}')
+        .to.equal('class Foo {$: string;} class Baz {$: string;}');
+  });
+});
+
+describe('Selective renaming', () => {
+  it('does not rename property names from the standard library', () => {
+    expectTranslate('Math.random();').to.equal('Math.random();');
+    expectTranslate('console.log();').to.equal('console.log();');
```

Comment:
I think one or two test cases (static method `Math.random` plus instance method `"foo".substring`, maybe plus a DOM method) would suffice here. Good testing is about testing different cases and situations, not just having a lot of test cases that essentially test the same.

Think about this as trying to find tests that cover each line of your code, each branch of an if statement, and (as much as possible) each permutation of the branches. You should have test cases to demonstrate each interesting branch works, but ideally only have exactly one test case - all other tests are just waste, as they do the same thing over and over again.